client/my-sites/site-settings/form-general.jsx

```diff
@@ -475,6 +478,182 @@ class SiteSettingsFormGeneral extends Component {
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Date Format' ) }
+				</FormLabel>
+				{ defaultFormats.map( ( format, key ) =>
+					<FormLabel key={ key }>
+						<FormRadio
+							checked={ format === date_format }
+							disabled={ isRequestingSettings }
+							name="date_format"
+							onChange={ handleRadio }
+							value={ format }
+						/>
+						<span>{ today.format( phpToMomentDatetimeFormat( format ) ) }</span>
+					</FormLabel>
+				) }
+				<FormLabel className="site-settings__date-time-format-custom">
+					<FormRadio
+						checked={ isCustomFormat }
+						disabled={ isRequestingSettings }
+						name="date_format"
+						onChange={ handleRadio }
+						value={ date_format }
+					/>
+					<span>
+						{ translate( 'Custom' ) }
+						<FormInput
+							disabled={ isRequestingSettings }
+							name="date_format_custom"
+							onChange={ onChangeField( 'date_format' ) }
+							type="text"
+							value={ date_format || '' }
+						/>
+						{ isCustomFormat && date_format
+							? today.format( phpToMomentDatetimeFormat( date_format ) )
+							: ''
+						}
+					</span>
+				</FormLabel>
+			</FormFieldset>
+		);
+	}
+
+	timeFormatOption() {
+		if ( ! config.isEnabled( 'manage/site-settings/date-time-format' ) ) {
+			return null;
+		}
+
+		const {
+			fields: { time_format, timezone_string },
+			handleRadio,
+			isRequestingSettings,
+			moment,
+			onChangeField,
+			translate,
+		} = this.props;
+
+		const defaultFormats = [ 'g:i a', 'g:i A', 'H:i' ];
+		const isCustomFormat = ! includes( defaultFormats, time_format );
+		const today = startsWith( timezone_string, 'UTC' )
+			? moment().utcOffset( timezone_string.substring( 3 ) * 60 )
+			: moment.tz( timezone_string );
+
+		return (
+			<FormFieldset>
+				<FormLabel>
+					{ translate( 'Time Format' ) }
+				</FormLabel>
+				{ defaultFormats.map( ( format, key ) =>
+					<FormLabel key={ key }>
+						<FormRadio
+							checked={ format === time_format }
+							disabled={ isRequestingSettings }
+							name="time_format"
+							onChange={ handleRadio }
+							value={ format }
+						/>
+						<span>{ today.format( phpToMomentDatetimeFormat( format ) ) }</span>
+					</FormLabel>
+				) }
+				<FormLabel className="site-settings__date-time-format-custom">
+					<FormRadio
+						checked={ isCustomFormat }
+						disabled={ isRequestingSettings }
+						name="time_format"
+						onChange={ handleRadio }
+						value={ time_format }
+					/>
+					<span>
+						{ translate( 'Custom' ) }
+						<FormInput
+							disabled={ isRequestingSettings }
+							name="time_format_custom"
+							onChange={ onChangeField( 'time_format' ) }
+							type="text"
+							value={ time_format || '' }
+						/>
+						{ isCustomFormat && time_format
+							? today.format( phpToMomentDatetimeFormat( time_format ) )
+							: ''
+						}
+					</span>
+					<FormSettingExplanation>
+						<a href="https://codex.wordpress.org/Formatting_Date_and_Time">
```

Comment:
Should we make this an [External Link](https://github.com/Automattic/wp-calypso/blob/master/client/components/external-link/README.md)?